packages/merge-on-green/src/merge-logic.ts

```diff
@@ -0,0 +1,538 @@
+          );
+          return mergeable;
+        }
+      } else if (checkCompleted.state !== 'success') {
+        console.info(
+          `Setting mergeable false due to ${checkCompleted.context} = ${checkCompleted.state}`
+        );
+        mergeable = false;
+        return mergeable;
+      }
+    }
+  } else {
+    mergeable = false;
+    console.log(
+      'Either you have no head sha, no required checks, or no MOG Label'
+    );
+    return mergeable;
+  }
+  return mergeable;
+};
+
+mergeOnGreen.getReviewsCompleted = async function getReviewsCompleted(
+  owner: string,
+  repo: string,
+  pr: number,
+  github: GitHubAPI
+): Promise<Reviews[]> {
+  try {
+    const reviewsCompleted = await github.pulls.listReviews({
+      owner,
+      repo,
+      pull_number: pr,
+    });
+    return reviewsCompleted.data;
+  } catch (err) {
+    return [];
+  }
+};
+
+mergeOnGreen.getReviewsRequested = async function getReviewsRequested(
+  owner: string,
+  repo: string,
+  pr: number,
+  github: GitHubAPI
+): Promise<ReviewsRequested> {
+  try {
+    return (
+      await github.pulls.listReviewRequests({
+        owner,
+        repo,
+        pull_number: pr,
+      })
+    ).data as ReviewsRequested;
+  } catch (err) {
+    return { users: [], teams: [] };
+  }
+};
+
+//this function cleans the reviews, since the listReviews method github provides returns a complete history of all comments added
+//and we just want the most recent for each reviewer
+mergeOnGreen.cleanReviews = function cleanReviews(
+  reviewsCompleted: Reviews[]
+): Reviews[] {
+  const cleanReviews = [];
+  const distinctReviewers: string[] = [];
+  for (let x = reviewsCompleted.length - 1; x >= 0; x--) {
+    const reviewsCompletedUser = reviewsCompleted[x].user.login;
+    if (!distinctReviewers.includes(reviewsCompletedUser)) {
+      cleanReviews.push(reviewsCompleted[x]);
+      distinctReviewers.push(reviewsCompletedUser);
+    }
+  }
+  return cleanReviews;
+};
+
+// TODO: test the logic for a team that's been assigned to review.
+// we might also want to make whether or not a requested viewer needs
+// to have been reviewed.
+mergeOnGreen.checkReviews = async function checkReviews(
+  owner: string,
+  repo: string,
+  pr: number,
+  github: GitHubAPI
+): Promise<boolean> {
+  console.info('=== checking required reviews ===');
+  let reviewsPassed = true;
+  const reviewsCompletedDirty = await mergeOnGreen.getReviewsCompleted(
+    owner,
+    repo,
+    pr,
+    github
+  );
+  const reviewsRequested = await mergeOnGreen.getReviewsRequested(
+    owner,
+    repo,
+    pr,
+    github
+  );
+  if (reviewsCompletedDirty.length !== 0) {
+    const reviewsCompleted = mergeOnGreen.cleanReviews(reviewsCompletedDirty);
+    if (reviewsCompleted.length !== 0) {
+      reviewsCompleted.forEach(review => {
+        if (review.state !== 'APPROVED') {
```

Comment:
this probably has an enum too

Reply:
I checked the return type, unfortunately it's a string. Good call out!